From 6128b26128d93c8b8cdbb75d10ef56fa9becf404 Mon Sep 17 00:00:00 2001
From: Weiguang Kong <weiguang.kong@nxp.com>
Date: Mon, 23 Oct 2017 09:36:04 +0800
Subject: [PATCH 2691/5242] MLK-16678-1: arm64: dts: distribute reserved
 memory for hifi4 dsp in SDRAM

commit  2aca97bd43e0ed413a5da076d3230363e01c6bf7 from
https://source.codeaurora.org/external/imx/linux-imx.git

Because the OCRAM memory size for hifi4 dsp is too small to keep its
code and data section, so distribute one reserved memory for hifi4 dsp
to save its code and data section in SDRAM, the address space that hifi4
can access in SDRAM is 0x81000000 - 0x9FFFFFFF, so the reserved memory
is as following:
   hifi4_reserved: hifi4@0x8e000000 {
       no-map;
       reg = <0 0x8e000000 0 0x1ffffff>;
   };

Signed-off-by: Weiguang Kong <weiguang.kong@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 arch/arm64/boot/dts/freescale/fsl-imx8qxp.dtsi |    5 +++++
 1 file changed, 5 insertions(+)

diff --git a/arch/arm64/boot/dts/freescale/fsl-imx8qxp.dtsi b/arch/arm64/boot/dts/freescale/fsl-imx8qxp.dtsi
index d24e2b9..bc6c7aa 100644
--- a/arch/arm64/boot/dts/freescale/fsl-imx8qxp.dtsi
+++ b/arch/arm64/boot/dts/freescale/fsl-imx8qxp.dtsi
@@ -64,6 +64,11 @@
 		#size-cells = <2>;
 		ranges;
 
+		hifi4_reserved: hifi4@0x8e000000 {
+			no-map;
+			reg = <0 0x8e000000 0 0x1ffffff>;
+		};
+
 		/* global autoconfigured region for contiguous allocations */
 		linux,cma {
 			compatible = "shared-dma-pool";
-- 
1.7.9.5

